### PR TITLE
Updated SQSEventQueueProvider options to be configurable. 

### PR DESCRIPTION
**Pull request by s3nthil**
- Updated SQSEventQueueProvider options to be configurable. 
- Added metric to track the rate at which event messages are handled.

### CODE REVIEW

File: contribs/src/main/java/com/netflix/conductor/core/events/sqs/SQSEventQueueProvider.java [hunk 1]
```diff
@@ -38,21 +39,27 @@
...
+	private final int pollTimeInMS;
+	private final int visibilityTimeout;
 	
 	@Inject
-	public SQSEventQueueProvider(AmazonSQSClient client) {
+	public SQSEventQueueProvider(AmazonSQSClient client, Configuration config) {
 		this.client = client;
+		this.batchSize = config.getIntProperty("workflow.event.queues.sqs.batchSize", 1);
+		this.pollTimeInMS = config.getIntProperty("workflow.event.queues.sqs.pollTimeInMS", 100);
+		this.visibilityTimeout = config.getIntProperty("workflow.event.queues.sqs.visibilityTimeout", 60);
```
Comment (cyzhao) [inline contribs/src/main/java/com/netflix/conductor/core/events/sqs/SQSEventQueueProvider.java]: consider name the property visibilityTimeoutSeconds.

File: contribs/src/main/java/com/netflix/conductor/core/events/sqs/SQSEventQueueProvider.java [hunk 2]
```diff
@@ -38,21 +39,27 @@
 
 	private final Map<String, ObservableQueue> queues = new ConcurrentHashMap<>();
 	private final AmazonSQSClient client;
+	private final int batchSize;
+	private final int pollTimeInMS;
+	private final int visibilityTimeout;
 	
 	@Inject
-	public SQSEventQueueProvider(AmazonSQSClient client) {
+	public SQSEventQueueProvider(AmazonSQSClient client, Configuration config) {
```
Comment (cyzhao) [inline contribs/src/main/java/com/netflix/conductor/core/events/sqs/SQSEventQueueProvider.java]: Add a unit test for this class.

File: core/src/main/java/com/netflix/conductor/metrics/Monitors.java [hunk 3]
```diff
@@ -232,6 +232,10 @@ public static void recordObservableQMessageReceivedErrors(String queueType) {
 		counter(classQualifier, "observable_queue_error", "queueType", queueType);
 	}
 
+	public static void recordEventQueueMessagesHandled(String queueType, String queueName) {
```
[the remark this answers is not in the document]
Comment (s3nthil) [reply core/src/main/java/com/netflix/conductor/metrics/Monitors.java]: The existing `recordEventQueueMessagesProcessed` metric only records the rate at which messages are pulled off the queue. This new metric `recordEventQueueMessagesHandler` records the rate at which the read messages are handled. Both are needed to find message throughput.

File: contribs/src/main/java/com/netflix/conductor/core/events/sqs/SQSEventQueueProvider.java (hunk 2 above)
Comment (s3nthil) [reply contribs/src/main/java/com/netflix/conductor/core/events/sqs/SQSEventQueueProvider.java]: Done

**Pull request merged**